runtime/topk.cu

```diff
@@ -38,28 +38,49 @@ struct CudaType {
   }
 };
 
+template <typename T>
```

Comment:
These type conversions between `__half` and `__nv_half` are rather annoying, but I couldn't find anything better. I think we should just use `<cuda_fp16.h>` instead of our own header. They are allowed to be redistributed, IIUC: https://docs.nvidia.com/cuda/eula/index.html#attachment-a